core/src/SYCL/Kokkos_SYCL_Space.cpp

```diff
@@ -63,10 +63,13 @@ void DeepCopySYCL(void* dst, const void* src, size_t n) {
 
 void DeepCopyAsyncSYCL(const Kokkos::Experimental::SYCL& instance, void* dst,
                        const void* src, size_t n) {
-  auto event =
-      instance.impl_internal_space_instance()->m_queue->memcpy(dst, src, n);
-  instance.impl_internal_space_instance()->m_queue->ext_oneapi_submit_barrier(
-      std::vector<sycl::event>{event});
+  // FIXME_SYCL using an in-order queue here should not be necessary since we
+  // are using submit_barrier for managing kernel dependencies but this seems to
+  // be required as a hot fix for now.
```

Comment:
Why do you mention "in-order queue" here?

Reply:
I just copy-pasted it without changing... Updated now.